File: Data Structures & Algorithms/daily-temperatures/submission-1.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        # Approach #1: runtime is O(n^2) and space is O(n)
        result = [0]*len(temperatures)
        stack = []
        for r in range(len(temperatures)):
            while stack and stack[-1][0] < temperatures[r]:
                idx = stack[-1][1]
                result[idx] = r-idx
                stack.pop()
            stack.append([temperatures[r],r]) 
        return result
        

File: Data Structures & Algorithms/count-paths/submission-1.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        vector<int> dp(n,1);
        for(int r = m-2; r >=0; r--){
            for(int c = n-2; c >= 0; c--){
               dp[c] += dp[c+1];
            }
        }
        return dp[0];   
    }
};
